libnet/src/ioctl.rs

```diff
@@ -26,25 +26,52 @@ use std::os::unix::io::AsRawFd;
 use std::ptr;
 use tracing::{debug, warn};
 
+// A wrapper to ensure that the argument passed to ioctl is mutable
+unsafe fn wrap_ioctl_rw<T>(fd: c_int, req: c_int, data: *mut T) -> c_int {
+    libc::ioctl(fd, req, data)
+}
+
 macro_rules! ioctl {
-    ( $fd:expr, $req:expr, $($args:expr),* ) => {{
-        match libc::ioctl($fd, $req, $($args),*) {
+    ( $fd:expr, $req:expr, $args:expr ) => {{
+        match wrap_ioctl_rw($fd, $req, $args) {
```

Comment:
This macro evaluates `$fd` twice. You could fix this by inserting a line above the `match`, e.g.

```rust
let fd = $fd;
```
...and then using `fd` below.

Reply:
Done